### PR TITLE
React select support in merge console

### PR DESCRIPTION
**Pull request by rehanalam**
### Reasons for making this change

DX Portal Pull Request -> https://github.com/apimatic/apimatic-dx-portal/pull/206

If this is related to existing tickets, include links to them as well.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've checked the rendering of the Markdown text I've added
  - [ ] If I'm adding a new section, I've updated the Table of Content
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated docs if needed
  - [ ] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: src/components/fields/DescriptionField.js [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
+
+  const renderers = {};
+  if (window.renderCodeBlock) {
+    renderers.code = window.renderCodeBlock;
   }
+
+  return (
+    <div id={id} className={pfx("field-description")}>
+      <ReactMarkdown
+        plugins={[]}
```
Comment (Inam-urRehman) [inline src/components/fields/DescriptionField.js]: do we need it with empty array?

File: src/components/fields/DescriptionField.js [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
+        children={description.replace(/<br>/gi, "\n")}
```
Comment (Inam-urRehman) [inline src/components/fields/DescriptionField.js:22]: children should be in child level

File: src/components/fields/MapField.js [hunk 3]
```diff
@@ -69,22 +69,34 @@ function DefaultMapItem(props) {
...
-    <div key={props.index} className={pfx(`row ${props.className}`)}>
-      {props.index > 0 && (
+    <div key={props.index} className={pfx(`${props.className}`)}>
+      {/* {props.index > 0 && (
         <div
           className="divider"
           style={{ borderTop: "1px solid #e2e5e7", margin: "30px 0" }}
         />
-      )}
+      )} */}
```
Comment (Inam-urRehman) [inline src/components/fields/MapField.js]: Remove commented code if useless

File: src/components/fields/MapField.js [hunk 4]
```diff
@@ -69,22 +69,34 @@ function DefaultMapItem(props) {
...
           alignItems: "center",
           justifyContent: "space-between"
         }}
       >
+        <IconBtn
+          tabIndex="-1"
+          onClick={() => props.toggleCollapse(props.key)}
+          className={pfx("btn toggle-button")}
+        >
+          {props.collapse ? (
```
Comment (Inam-urRehman) [inline src/components/fields/MapField.js:87]: condition should be in rotate prop e.g `props.collapse ? 90: 0`

File: src/components/fields/MapField.js [hunk 5]
```diff
@@ -103,42 +115,75 @@ function DefaultMapItem(props) {
...
-          >
-            {props.children}
+      {!props.collapse && (
+        <div className={pfx("map-field-value-container")}>
+          <div className={pfx("flex")} style={{ display: "flex" }}>
+            <div
+              className={pfx(
+                `flex-1 ${props.hasToolbar && props.hasRemove ? "" : ""}`
+              )}
+              style={{ flex: "1" }}
```
Comment (Inam-urRehman) [inline src/components/fields/MapField.js:123]: avoid inline styling

File: src/components/fields/MapField.js [hunk 6]
```diff
@@ -103,42 +115,75 @@ function DefaultMapItem(props) {
...
+            disabled={props.disabled}
+          />
+        </div>
+
+        <IconBtn
+          tabIndex="-1"
+          onClick={props.toggleGroupCollapse}
+          className={pfx("btn toggle-button")}
+        >
+          {props.collapse ? (
```
Comment (Inam-urRehman) [inline src/components/fields/MapField.js:162]: same as above

File: src/components/fields/MapField.js [hunk 7]
```diff
@@ -183,10 +232,26 @@ class MapField extends Component {
...
-    this.setState(this.getStateFromProps(nextProps));
+    const collapse =
+      this.state.depth === this.state.expandAllLevel &&
+      this.state.expandAll !== nextProps.expandAll
+        ? !nextProps.expandAll
+        : this.state.collapse;
+    this.setState({
+      ...this.getStateFromProps(nextProps),
+      collapse,
+      expandAllLevel: this.state.expandAllLevel,
```
Comment (Inam-urRehman) [inline src/components/fields/MapField.js:248]: use `this.setState( st => ({...st, ...this.getStateFromProps(nextPorps), collapse, expandAll: ..})`

File: src/components/fields/MapField.js [hunk 8]
```diff
@@ -403,6 +488,8 @@ class MapField extends Component {
           definitions,
           item
         );
+
+        const collapse = this.state.expandInfo[pair.k] ? false : true;
```
Comment (Inam-urRehman) [inline src/components/fields/MapField.js:490]: You can use this for code readability `Boolean(this.state.expandInfo[pair.k])`

File: src/components/fields/ObjectField.js [hunk 9]
```diff
@@ -86,29 +114,95 @@ function DefaultObjectFieldTemplate(props) {
...
+
+        {canEditJson && renderViewJsonButton(props)}
+
+        {canCollapse && (
+          <IconBtn
+            tabIndex="-1"
+            onClick={toggleCollapse}
+            className={pfx("btn toggle-button")}
+          >
+            {collapse ? (
```
Comment (Inam-urRehman) [inline src/components/fields/ObjectField.js:178]: same as above

File: src/components/fields/ObjectField.js [hunk 10]
```diff
@@ -86,29 +114,95 @@ function DefaultObjectFieldTemplate(props) {
...
+              <ChevronIcon width={14} rotate={-90} />
+            ) : (
+              <ChevronIcon width={14} />
+            )}
+          </IconBtn>
+        )}
+      </div>
+
+      {/* {dataType && (
```
Comment (Inam-urRehman) [inline src/components/fields/ObjectField.js]: remove commented code

File: src/components/fields/ObjectField.js [hunk 11]
```diff
@@ -159,11 +261,26 @@ class ObjectField extends Component {
...
-    this.setState(this.getStateFromProps(nextProps));
+    const collapse =
+      this.state.depth === this.state.expandAllLevel &&
+      this.state.expandAll !== nextProps.expandAll
+        ? !nextProps.expandAll
+        : this.state.collapse;
+    this.setState({
+      ...this.getStateFromProps(nextProps),
+      collapse,
+      expandAllLevel: this.state.expandAllLevel,
```
Comment (Inam-urRehman) [inline src/components/fields/ObjectField.js:274]: same as above

File: src/components/widgets/SelectWidget.js [hunk 12]
```diff
@@ -53,44 +69,40 @@ function SelectWidget(props) {
   } = props;
   const { enumOptions, enumDisabled } = options;
   const emptyValue = multiple ? [] : "";
+  let newOptions = enumOptions.map(o => {
```
Comment (Inam-urRehman) [inline src/components/widgets/SelectWidget.js:72]: don't use let unless you need it

File: src/components/widgets/SelectWidget.js [hunk 13]
```diff
@@ -53,44 +69,40 @@ function SelectWidget(props) {
...
+    <Select
       id={id}
-      multiple={multiple}
       className={pfx("form-control")}
-      value={typeof value === "undefined" ? emptyValue : value}
+      classNamePrefix="react-select"
+      isMulti={multiple}
+      options={newOptions}
+      defaultValue={
+        typeof value === "undefined"
```
Comment (Inam-urRehman) [inline src/components/widgets/SelectWidget.js]: `value ? getDefaultValue(): emptyValue`

File: src/components/widgets/SelectWidget.js [hunk 14]
```diff
@@ -119,3 +131,43 @@ if (process.env.NODE_ENV !== "production") {
 }
 
 export default SelectWidget;
+
+{
```
Comment (Inam-urRehman) [inline src/components/widgets/SelectWidget.js]: Remove commented

File: src/components/widgets/SelectWidget.js (hunk 13 above)
Comment (rehanalam) [reply src/components/widgets/SelectWidget.js]: done

File: src/components/fields/MapField.js (hunk 3 above)
Comment (rehanalam) [reply src/components/fields/MapField.js]: done

File: src/components/fields/MapField.js (hunk 5 above)
Comment (rehanalam) [reply src/components/fields/MapField.js:123]: Existing issue. For now skipping it.

File: src/components/fields/MapField.js (hunk 4 above)
Comment (rehanalam) [reply src/components/fields/MapField.js:87]: Exisitng issue. Will required additional testing. For now skipping it.

File: src/components/fields/MapField.js (hunk 8 above)
Comment (rehanalam) [reply src/components/fields/MapField.js:490]: Existing issue. Dont want to touch previous logic becuase it reqiures additional testing

File: src/components/fields/MapField.js (hunk 7 above)
Comment (rehanalam) [reply src/components/fields/MapField.js:248]: Existing issue. Dont want to touch previous logic becuase it reqiures additional testing

File: src/components/fields/ObjectField.js (hunk 9 above)
Comment (rehanalam) [reply src/components/fields/ObjectField.js:178]: Existing issue. Dont want to touch previous logic becuase it reqiures additional testing.

Comment (rehanalam) [reply src/components/fields/ObjectField.js:178]: done

**Pull request merged**